From f7b5a0d483e64bc81cd62e8a93f296abfe508f53 Mon Sep 17 00:00:00 2001
From: Artur Gebicz <a.gebicz@metrological.com>
Date: Fri, 4 Dec 2020 07:24:15 +0000
Subject: [PATCH] [monitor] Enable monitored plugins when built from external
 repositories

Previously the monitored plugins were enabled autmatically when the
plugin was enabled. However due to split of plugins betweeen
ThunderNanoServices and RDKServices, some of the plugins resides in
different repository, thus cannot be autmatically enabled.
This commit introduces the possibility to automatically monitor
plugins that are built within this repository and manually enable
plugins that are outside of this repository.
---
 Monitor/CMakeLists.txt | 31 +++++++++++++++++++++++--------
 Monitor/Monitor.config | 34 +++++++++++++++++-----------------
 2 files changed, 40 insertions(+), 25 deletions(-)

diff --git a/Monitor/CMakeLists.txt b/Monitor/CMakeLists.txt
index 3453245..77a7adb 100644
--- a/Monitor/CMakeLists.txt
+++ b/Monitor/CMakeLists.txt
@@ -18,14 +18,29 @@
 set(PLUGIN_NAME Monitor)
 set(MODULE_NAME ${NAMESPACE}${PLUGIN_NAME})
 
-set(PLUGIN_WEBKITBROWSER_MEMORYLIMIT "614400" CACHE STRING "monitor webkit memory limit")
-set(PLUGIN_YOUTUBE_MEMORYLIMIT "614400" CACHE STRING "monitor youtube memory limit")
-set(PLUGIN_COBALT_MEMORYLIMIT "614400" CACHE STRING "monitor cobalt memory limit")
-set(PLUGIN_AMAZON_MEMORYLIMIT "614400" CACHE STRING "monitor amazon  memory limit")
-set(PLUGIN_APPS_MEMORYLIMIT "614400" CACHE STRING "monitor apps memory limit")
-set(PLUGIN_UX_MEMORYLIMIT "614400" CACHE STRING "monitor ux memory limit")
-set(PLUGIN_NETFLIX_MEMORYLIMIT "614400" CACHE STRING "monitor netflix memory limit")
-set(PLUGIN_RESIDENT_APP_MEMORYLIMIT "614400" CACHE STRING "monitor resident app memory limit")
+# Plugins built from this repository that can be autmatically enabled or enabled manually when built externally
+set(PLUGIN_MONITOR_WEBKITBROWSER "${PLUGIN_WEBKITBROWSER}" CACHE BOOL "Enable monitor for the WebKitBrowser plugin")
+set(PLUGIN_MONITOR_WEBKITBROWSER_APPS "${PLUGIN_WEBKITBROWSER_APPS}" CACHE BOOL "Enable monitor for the WebKitBrowser Apps plugin")
+set(PLUGIN_MONITOR_WEBKITBROWSER_RESIDENT_APP "${PLUGIN_WEBKITBROWSER_RESIDENT_APP}" CACHE BOOL "Enable monitor for the WebKitBrowser Resident App plugin")
+set(PLUGIN_MONITOR_WEBKITBROWSER_UX "${PLUGIN_WEBKITBROWSER_UX}" CACHE BOOL "Enable monitor for the WebKitBrowser UX plugin")
+set(PLUGIN_MONITOR_WEBKITBROWSER_YOUTUBE "${PLUGIN_WEBKITBROWSER_YOUTUBE}" CACHE BOOL "Enable monitor for the WebKitBrowser Youtube plugin")
+set(PLUGIN_MONITOR_OPENCDMI "${PLUGIN_OPENCDMI}" CACHE BOOL "Enable monitor for the OpenCDMI plugin")
+
+# Plugins built from outside of this repository have to be enabled manually
+set(PLUGIN_MONITOR_COBALT OFF CACHE BOOL "Enable monitor for the Cobalt plugin")
+set(PLUGIN_MONITOR_NETFLIX OFF CACHE BOOL "Enable monitor for the Netflix plugin")
+set(PLUGIN_MONITOR_OUTOFPROCESS OFF CACHE BOOL "Enable monitor for the OutOfProcess plugin")
+set(PLUGIN_MONITOR_TESTUTILITY OFF CACHE BOOL "Enable monitor for the TestUtility plugin")
+
+# Options for each plugin
+set(PLUGIN_MONITOR_WEBKITBROWSER_MEMORYLIMIT "614400" CACHE STRING "monitor webkit memory limit")
+set(PLUGIN_MONITOR_YOUTUBE_MEMORYLIMIT "614400" CACHE STRING "monitor youtube memory limit")
+set(PLUGIN_MONITOR_COBALT_MEMORYLIMIT "614400" CACHE STRING "monitor cobalt memory limit")
+set(PLUGIN_MONITOR_AMAZON_MEMORYLIMIT "614400" CACHE STRING "monitor amazon  memory limit")
+set(PLUGIN_MONITOR_APPS_MEMORYLIMIT "614400" CACHE STRING "monitor apps memory limit")
+set(PLUGIN_MONITOR_UX_MEMORYLIMIT "614400" CACHE STRING "monitor ux memory limit")
+set(PLUGIN_MONITOR_NETFLIX_MEMORYLIMIT "614400" CACHE STRING "monitor netflix memory limit")
+set(PLUGIN_MONITOR_RESIDENT_APP_MEMORYLIMIT "614400" CACHE STRING "monitor resident app memory limit")
 
 find_package(${NAMESPACE}Plugins REQUIRED)
 find_package(CompileSettingsDebug CONFIG REQUIRED)
diff --git a/Monitor/Monitor.config b/Monitor/Monitor.config
index 147e2bb..af6fb97 100644
--- a/Monitor/Monitor.config
+++ b/Monitor/Monitor.config
@@ -4,11 +4,11 @@ map()
 end()
 ans(configuration)
 
-if(PLUGIN_WEBKITBROWSER)
+if(PLUGIN_MONITOR_WEBKITBROWSER)
     map()
         kv(callsign WebKitBrowser)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_WEBKITBROWSER_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_WEBKITBROWSER_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
@@ -21,11 +21,11 @@ if(PLUGIN_WEBKITBROWSER)
     map_append(${configuration} observables ${WEBKITBROWSER_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_WEBKITBROWSER_YOUTUBE)
+if(PLUGIN_MONITOR_WEBKITBROWSER_YOUTUBE)
     map()
         kv(callsign YouTube)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_COBALT_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_COBALT_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
@@ -38,11 +38,11 @@ if(PLUGIN_WEBKITBROWSER_YOUTUBE)
     map_append(${configuration} observables ${YOUTUBE_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_COBALT)
+if(PLUGIN_MONITOR_COBALT)
     map()
         kv(callsign Cobalt)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_YOUTUBE_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_YOUTUBE_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
@@ -55,11 +55,11 @@ if(PLUGIN_COBALT)
     map_append(${configuration} observables ${COBALT_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_WEBKITBROWSER_APPS)
+if(PLUGIN_MONITOR_WEBKITBROWSER_APPS)
     map()
         kv(callsign Apps)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_APPS_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_APPS_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
@@ -72,11 +72,11 @@ if(PLUGIN_WEBKITBROWSER_APPS)
     map_append(${configuration} observables ${APPS_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_WEBKITBROWSER_UX)
+if(PLUGIN_MONITOR_WEBKITBROWSER_UX)
     map()
         kv(callsign UX)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_UX_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_UX_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
@@ -89,11 +89,11 @@ if(PLUGIN_WEBKITBROWSER_UX)
     map_append(${configuration} observables ${UX_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_NETFLIX)
+if(PLUGIN_MONITOR_NETFLIX)
     map()
         kv(callsign Netflix)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_NETFLIX_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_NETFLIX_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
@@ -106,7 +106,7 @@ if(PLUGIN_NETFLIX)
     map_append(${configuration} observables ${NETFLIX_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_OUTOFPROCESS)
+if(PLUGIN_MONITOR_OUTOFPROCESS)
     map()
         kv(callsign OutOfProcessPlugin)
         kv(memory 5)
@@ -123,7 +123,7 @@ if(PLUGIN_OUTOFPROCESS)
     map_append(${configuration} observables ${OUTOFPROCESSTEST_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_TESTUTILITY)
+if(PLUGIN_MONITOR_TESTUTILITY)
     map()
         kv(callsign TestUtility)
         kv(memory 5)
@@ -140,7 +140,7 @@ if(PLUGIN_TESTUTILITY)
     map_append(${configuration} observables ${TESTUTILITY_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_OPENCDMI)
+if(PLUGIN_MONITOR_OPENCDMI)
     map()
         kv(callsign OCDM)
         kv(operational 1)
@@ -155,11 +155,11 @@ if(PLUGIN_OPENCDMI)
     map_append(${configuration} observables ${OPENCDMI_MONITOR_CONFIG})
 endif()
 
-if(PLUGIN_WEBKITBROWSER_RESIDENT_APP)
+if(PLUGIN_MONITOR_WEBKITBROWSER_RESIDENT_APP)
     map()
         kv(callsign ResidentApp)
         kv(memory 5)
-        kv(memorylimit ${PLUGIN_RESIDENT_APP_MEMORYLIMIT})
+        kv(memorylimit ${PLUGIN_MONITOR_RESIDENT_APP_MEMORYLIMIT})
         kv(operational 1)
         key(restart)
         map()
-- 
2.28.0

